cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -255,4 +258,19 @@ public DatasetAdmin call() throws Exception {
       }
     }
   }
+
+  private static UserGroupInformation getUgiForDataset(Impersonator impersonator, DatasetId datasetInstanceId)
+    throws IOException, NamespaceNotFoundException {
+    // for system dataset do not look up owner information in store as we know that it will be null.
+    // Also, this is required for CDAP to start, because initially we don't want to look up owner admin
+    // (causing its own lookup) as the SystemDatasetInitiator.getDataset is called when CDAP starts
+    UserGroupInformation ugi;
```

Comment:
Without this I was CDAP was stuck in a loop during startup. We were looking for system dataset owners while the owner admin dataset was not ready. If you made a change to fix this cycle then I can this won't be needed.